src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * ZookeeperStore is a StoreInterface implementation using Zookeeper as store engine.
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ */
+class ZookeeperStore implements StoreInterface
+{
+    /**
+     * @var \Zookeeper
+     */
+    private $zookeeper;
+
+    /**
+     * Locker constructor.
+     *
+     * @param \Zookeeper $zookeeper Zookeeper Instance
+     */
+    public function __construct(\Zookeeper $zookeeper)
+    {
+        $this->zookeeper = $zookeeper;
+    }
+
+    /**
+     * Stores the resource if it's not locked by someone else.
+     *
+     * @param Key $key the Key that represent the lock resource
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public function save(Key $key)
+    {
+        $resource = $this->getKeyResource($key);
+
+        // This makes sure the node dies at the end of a client session, thus releasing the lock to avoid starvation
+        $flags = \Zookeeper::EPHEMERAL;
+
+        try {
+            $token = $this->getToken($key);
+
+            if ($this->exists($key)) {
+                if ($token === $this->zookeeper->get($resource)) {
+                    return;
+                }
+            }
+
+            $this->createNewLock($resource, $token, array(), $flags);
+        } catch (LockConflictedException $ex) {
+            throw $ex;
+        } catch (\Throwable $throwable) {
+            throw new LockAcquiringException($throwable);
+        }
```

Comment:
According to method `errorHandlerForCreateNode()`, I think you should catch a `ZookeeperException` here.
And if the `$ze->code` equals to `Zookeeper::NODEEXISTS`, it should throw `LockConflictedException` instead of `LockAcquiringException`.

I'm not familiar with Symfony.
So if I'm wrong, just forget about it.
:)

Reply:
Thanks for the suggestion. Can you share the documentation around PHP Zookeeper Exceptions? I couldn't find any details related to that so didn't know this was a possibility. Would love to look into it.

Reply:
Removed error handler as I developed this initially against `0.2.3` version of the extension which didn't throw exceptions.

Reply:
Sorry, it's my fault.
I'll fix the errors in http://php.net/manual/en/book.zookeeper.php ASAP.
: (